### PR TITLE
Updated VIMVideo SVOD Extension

### PR DESCRIPTION
**Pull request by jenniferlim**
#### Ticket

N/A

#### Issue Summary

- We didn't return an invalid type for SVODAccess if the video isn't SVOD.

#### Implementation Summary

- Created a new type Invalid for svodAccess. That way if a video isn't svod and we try to access it, it will return an invalid type.

#### How to Test

N/A

### CODE REVIEW

**Comment by jenniferlim** [comment]
Build this please

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          4 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/d9500396053b6f2f847f5687d712b99ebadd2c77/VimeoNetworking/Sources/Request+Notifications.swift#L54">VimeoNetworking/Sources/Request+Notifications.swift#L54</a></strong>: result of call to ‘map’ is unused  <br /><code>
        notifications.map { (notification: VIMNotification) -&gt; Void in
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/d9500396053b6f2f847f5687d712b99ebadd2c77/VimeoNetworking/Sources/VimeoClient.swift#L328">VimeoNetworking/Sources/VimeoClient.swift#L328</a></strong>: cast from ‘Response&lt;VIMNullResponse&gt;’ to unrelated type ‘Response&lt;ModelType&gt;’ always fails  <br /><code>
                let response = Response(model: nullResponseObject, json: [:]) as! Response&lt;ModelType&gt;
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/d9500396053b6f2f847f5687d712b99ebadd2c77/VimeoNetworking/Sources/Models/VIMVideoPreference.m#L41">VimeoNetworking/Sources/Models/VIMVideoPreference.m#L41</a></strong>: incompatible pointer types assigning to ‘NSString * _Nullable’ from ‘VIMPrivacy * _Nullable’ [-Wincompatible-pointer-types]  <br /><code>
        privacy.view = self.privacy;
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/d9500396053b6f2f847f5687d712b99ebadd2c77/VimeoNetworking/Sources/Models/VIMVODItem.m#L285">VimeoNetworking/Sources/Models/VIMVODItem.m#L285</a></strong>: incompatible pointer types returning ‘VIMConnection * _Nullable’ from a function with result type ‘VIMVODConnection *’ [-Wincompatible-pointer-types]  <br /><code>
    return [self connectionWithName:VIMConnectionNameVideos];
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 22 tests, with 0 failures (0 unexpected) in 0.268 (0.294) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoNetworking/243)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: VimeoNetworking/Sources/Models/VIMVideo+SVOD.m [hunk 1]
```diff
@@ -16,7 +16,12 @@ - (BOOL)isSVOD
 {
     VIMInteraction *interation = [self interactionWithName:VIMInteractionNameSVOD];
     
-    return interation != nil;
+    if (interation != nil)
```
Comment (mikew-personal) [inline VimeoNetworking/Sources/Models/VIMVideo+SVOD.m]: why not leave this as return interaction != nil; ?  It seems like less code.

File: VimeoNetworking/Sources/Models/VIMVideo+SVOD.m [hunk 2]
```diff
@@ -35,9 +40,12 @@ - (VIMVideoSVODAccess)svodAccess
         }
         else
         {
-            return VIMVideoSVODAccessUnknown;
+            return VIMVideoSVODAccessInvalid;
         }
     }
+    
+    return VIMVideoSVODAccessNotSVOD;
```
Comment (mikew-personal) [inline VimeoNetworking/Sources/Models/VIMVideo+SVOD.m]: 👍

**Pull request merged**